tests/public-api/examples/optimization/solver.py

```diff
@@ -0,0 +1,194 @@
+            self._solver = self._solvers_api.get_solver_release(self._solver_key, self._solver_version)
+            self._job = self._solvers_api.create_job( self._solver.id, self._solver.version, JobInputs({"input_1": input}))
+
+            # solve
+            logging.info(f'Start solving job: {self._job.id}')
+
+            self._status = self._solvers_api.start_job(self._solver.id, self._solver.version, self._job.id)
+        except osparc.exceptions.ApiException as e:
+            self._api_client.close()
+            expt = json.loads(e.body)
+            raise Exception('\n'.join(expt["errors"])) from None
+                
+    def job_done(self) -> bool:
+        """
+        Check if a submitted job is done
+        """
+        try:
+            if self._job is None:
+                return True
+            self._status = self._solvers_api.inspect_job(self._solver_key, self._solver_version, self._job.id)
+            if self._status.stopped_at:
+                if self._status.state != 'SUCCESS':
+                    logging.error(f'Failed job {self._job.id} with status {self._status.state}')
+                    log: List[str] = [f'Failed to solve job with status {self._status.state}', 'Server log:']
+                    log += self._generate_isolve_log()
+                    raise Exception('\n'.join(log))
+                return True
+            else:
+                return False
+        except osparc.exceptions.ApiException as e:
+            self._api_client.close()
+            expt = json.loads(e.body)
+            raise Exception('\n'.join(expt["errors"])) from None           
+        
+    def fetch_results(self, output_file: Path, log_path: Path) -> bool:
+        """
+        Fetches the results of a simulation
+        """
+        try:
+            outputs: JobOutputs = self._solvers_api.get_job_outputs(self._solver_key, self._solver_version, self._job.id)
+            for _, result in outputs.results.items():
+                file = self._files_api.download_file(result.id)
+                if result.filename == 'output.h5':
+                    shutil.move(file, output_file)
+                elif result.filename == 'log.tgz':
+                    shutil.move(file, log_path)
+                else:
+                    os.remove(file)
+                    logging.info(f'Received unexpected output: {result.filename}')
+                    continue
+                logging.info(f'Successfully downloaded {result.filename}')
+                
+            self._job = None
+            self._status = None
+        except osparc.exceptions.ApiException as e:
+            self._api_client.close()
+            expt = json.loads(e.body)
+            raise Exception('\n'.join(expt["errors"])) from None
+
+
+
+def call_osparc_solver(solver_key: str, solver_version: str, input_file: Path, output_file: Path, log_path: Path, cfg: Optional[osparc.Configuration] = None) -> None:
```

Comment:
This line is rather long - again I guess `black` would resolve this

Reply:
yep, I will try to run black on these files before merging